### PR TITLE
Update SDDP 

### PR DESCRIPTION
**Pull request by frapac**
Updates:
- increase generic code
- add piecewise linear costs to model
- add stopping criteria (with Monte-Carlo simulation)

### CODE REVIEW

**Comment by odow** [comment]
Just wondering why you need `push!(LOAD_PATH, "../src")` at the top of your examples?

File: examples/damsvalley.jl [hunk 1]
```diff
@@ -7,42 +7,51 @@
...
-N_STAGES = 52
-N_SCENARIOS = 2
+alea_year = Array([7.0 7.0 8.0 3.0 1.0 1.0 3.0 4.0 3.0 2.0 6.0 5.0 2.0 6.0 4.0 7.0 3.0 4.0 1.0 1.0 6.0 2.0 2.0 8.0 3.0 7.0 3.0 1.0 4.0 2.0 4.0 1.0 3.0 2.0 8.0 1.0 5.0 5.0 2.0 1.0 6.0 7.0 5.0 1.0 7.0 7.0 7.0 4.0 3.0 2.0 8.0 7.0])
+
+const N_STAGES = 52
+const N_SCENARIOS = 10
 
 # FINAL TIME:
-TF = 52
+const TF = 52
```
Comment (Henri-Gerard) [inline examples/damsvalley.jl:27]: Why use TF since it equals to N_STAGES

File: test/runtests.jl [hunk 2]
```diff
@@ -6,6 +6,8 @@
 # run unit-tests
 #############################################################################
 
+push!(LOAD_PATH, "src")
```
Comment (Henri-Gerard) [inline test/runtests.jl:9]: Great ! It solves the problem of loading the files in /src

Comment (odow) [inline test/runtests.jl:9]: Is all the code under the `~/.julia/v0.4/StochDynamicProgramming/` folder? You shouldn't need this line. `using StochDynamicProgramming` should be sufficient. Then you can either reference unexported methods with the `StochDynamicProgramming.` prefix or export them.

**Pull request merged**